backend/compact-connect/lambdas/python/common/cc_common/data_model/schema/user.py

```diff
@@ -18,6 +18,19 @@ class CompactPermissionsRecordSchema(Schema):
         allow_none=False,
     )
 
+    @post_dump
+    def drop_empty_actions(self, data, **kwargs):  # noqa: ARG002 unused-kwargs
+        """
+        DynamoDB doesn't like empty sets, so we will make a point to drop an actions field entirely,
+        if it is empty.
+        """
+        if not data.get('actions', {}):
+            data.pop('actions', None)
+        empty_jurisdictions = [jurisdiction for jurisdiction, actions in data['jurisdictions'].items() if not actions]
+        for jurisdiction in empty_jurisdictions:
+            del data['jurisdictions'][jurisdiction]
```

Comment:
For consistency, can we do this?
```suggestion
           data['jurisdictions'].pop(jurisdiction, None)
```

Reply:
We could, though they are different because we are expecting different behaviors there:
`data.pop(X, None)` will not produce a `KeyError` in the event that `X` is not in `data`, but `del data[X]` will. I'm effectively using `data.pop(X, None)` as a more concise way of doing:
```python
try:
    del data[X]
except KeyError:
    pass
```

I'd rather not suppress a `KeyError` in the latter call because I expect that the key _should_ be in the `data` object and it would be a bug if I were wrong.

Reply:
Ah that makes sense. Thanks for clarifying!